File: backend/src/MessagingPlatform.Application/Common/Models/MessageDTOs/MessageDto.cs
using System.ComponentModel.DataAnnotations;

namespace MessagingPlatform.Application.Common.Models.MessageDTOs;

public class MessageDto
{
    public required string SenderFullName { get; init; }
    public required string Content { get; set; }
    
    public required DateTime SentAt { get; init; }
    public DateTime? UpdatedAt { get; set; }
    public bool IsRead { get; set; } = false;
}

File: frontend/src/app/models/messagedto.ts
import { UserDto } from "./userdto";

export interface MessageDto {
  id: string;
  senderId: string;
  sender: UserDto | null;
  chatId: string;
  content: string;
  sentAt: Date;
  updatedAt: Date | null;
  isRead: boolean;
}
